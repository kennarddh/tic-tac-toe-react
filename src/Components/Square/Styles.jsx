import styled from 'styled-components'

export const Container = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	padding: 10px;
	width: 100px;
	height: 100px;
	background-color: #f5f5f5;
	box-sizing: border-box;
	border: 1px solid #ccc;
`
