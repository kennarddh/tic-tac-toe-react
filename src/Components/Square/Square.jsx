import React, { memo } from 'react'

// Styles
import { Container } from './Styles'

const Square = ({ value, onClick, index }) => {
	return (
		<Container
			onClick={() => onClick(index)}
			data-testid={`square-${index}`}
		>
			{value}
		</Container>
	)
}

export default memo(Square)
