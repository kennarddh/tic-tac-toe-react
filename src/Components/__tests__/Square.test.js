import React from 'react'
import Square from '../Square/Square'

import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('This suit is to test the Square component', () => {
	test('Snapshot of Square', () => {
		const { asFragment } = render(<Square value='1' />)

		expect(asFragment()).toMatchSnapshot()
	})

	test('Check square rendering value', () => {
		const { getByText } = render(<Square value='1' />)

		expect(getByText('1')).toBeInTheDocument('1')
	})

	test('Check click event', () => {
		const onClick = jest.fn()
		const { getByText } = render(<Square value='1' onClick={onClick} />)

		userEvent.click(getByText('1'))

		expect(onClick).toHaveBeenCalledTimes(1)
	})

	test('Check click event with value', () => {
		const onClick = jest.fn()
		const { getByText } = render(<Square value='1' index='1' onClick={onClick} />)

		userEvent.click(getByText('1'))

		expect(onClick).toHaveBeenCalledWith('1')
	})

	test('Check click event with value and index', () => {
		const onClick = jest.fn()
		const { getByText } = render(<Square value='1' index='1' onClick={onClick} />)

		userEvent.click(getByText('1'))

		expect(onClick).toHaveBeenCalledTimes(1)
		expect(onClick).toHaveBeenCalledWith('1')
	})
})
