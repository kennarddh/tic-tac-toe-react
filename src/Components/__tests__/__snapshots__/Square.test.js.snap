// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the Square component Snapshot of Square 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ gkhTDj"
    data-testid="square-container"
  >
    1
  </div>
</DocumentFragment>
`;
